render/vnode.js

```diff
@@ -8,9 +8,10 @@ Vnode.normalize = function(node) {
 	if (node != null && typeof node !== "object") return Vnode("#", undefined, undefined, node === false ? "" : node, undefined, undefined)
 	return node
 }
-Vnode.normalizeChildren = function normalizeChildren(children) {
-	for (var i = 0; i < children.length; i++) {
-		children[i] = Vnode.normalize(children[i])
+Vnode.normalizeChildren = function normalizeChildren(input) {
+	var children = []
```

Comment:
The issue with having non-normalized children *and* normalized instance children is that up until now, there was no need to distinguish between the two so they were effectively the same thing. New arrays here stops downstream code mutating upstream references, which is otherwise pretty unsettling.

Reply:
That's a good point, but I'm afraid it will not be good for performance.

`vnodes` should IMO be treated as owned by `render` once they leave user code.

OTOH having your children turn from `["a", "b", "c"]` to an array of `vnodes` can definitely be surprising.

I'm on the fence on this...